File: Homeworks/Homework1/Calculator.cs
using System;

namespace Homework1
{
    public enum Operations
    {
        Plus,Minus,Multiply,Divide,Unknown  
    }
    public static class Calculator
    {
        public static double Calculate(int val1, Operations operation, int val2)
        {
            return operation switch
            {
                Operations.Plus => val1 + val2,
                Operations.Minus => val1 - val2,
                Operations.Multiply => val1 * val2,
                Operations.Divide => (double)val1 / val2
            };
        }
    }
}

File: Homeworks/Homework4/Homework4/Calculator.fs
module Homework4.Calculator
open System
let Calculate val1 operation val2 =
    if operation = Operation.Divide && val2 = 0. then
        raise (DivideByZeroException $"can`t divide by zero!")
    let result =
        match operation with
        | Plus -> val1 + val2
        | Minus -> val1 - val2
        | Multiply -> val1 * val2
        | Divide -> (double)val1 / val2
        | Unknown -> raise (ArgumentException $"{operation}is not correct argument")
    result     